### ISSUE TITLE
Umem::new() failing on latest kernels

### ISSUE THREAD

**Issue by astoycos** [HeroicKatora/xdpilone#3]
It seems like 

`unsafe { Umem::new(UmemConfig::default(), mem).unwrap() }` 

Is resulting in the following failure 

```
thread 'tests::xdp::af_xdp' panicked at test/integration-test/src/tests/xdp.rs:40:56:
called `Result::unwrap()` on an `Err` value: Errno(22: Invalid argument)
```

and the strace looks like so...

```
126680 setsockopt(13, SOL_XDP, XDP_UMEM_REG, "\0@\213\234\336\177\0\0\0\20\0\0\0\0\0\0\0\20\0\0\0\0\0\0\0\0\0\0\0\0\0\200", 32) = -1 EINVAL (Invalid argument)
```

You can easily repro locally with aya's CI on a newer kernel or see in gh actions in https://github.com/aya-rs/aya/actions/runs/10458255089/job/28959641356?pr=1013

**Comment by HeroicKatora** [comment]
What's the `latest' kernel you're referring to exactly? Oddly enough the `flood-mt` example still works just fine on my machine.

```
Linux Chimera 6.10.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 27 Jul 2024 16:49:55 +0000 x86_64 GNU/Linux
```

**Comment by HeroicKatora** [comment]
Something is seriously weird. My UB meters are tingling. It makes a difference whether the `umem.rs:91` check is unwrapped or checked with `?`. The latter results in the error as in the bug report, the former just .. appears to work. Will have a closer look tomorrow I suppose.

**Comment by HeroicKatora** [comment]
Preliminary idea: the `config` might have gained additional fields on the kernel side. It has, `	__u32 tx_metadata_len;
` is not in my struct definition from 6.6. (The `\200` in your strace might be that garbage field). If that's the case it's not full UB then but definitely unintended. But I'm passing `optlen` according to the libraries definition of this struct. Maybe the kernel is ignoring this?

**Comment by HeroicKatora** [comment]
Oh. No. It's padding bytes in this lib's definition, now the kernel has a field that it interprets. This is a bug.

**Comment by HeroicKatora** [comment]
Fixed in `1.0.5`.

**Comment by astoycos** [comment]
Ahh thanks for the super quick response @HeroicKatora!
